packages/flutter/test/rendering/semantics_test.dart

```diff
@@ -104,10 +105,105 @@ void main() {
+    );
+
+    expect(
+      root.toStringDeep(childOrder: DebugSemanticsDumpOrder.inverseHitTest),
+      'SemanticsNode#11(stale, owner: null, Rect.fromLTRB(0.0, 0.0, 0.0, 0.0))\n'
+          '├SemanticsNode#9(stale, owner: null, Rect.fromLTRB(20.0, 20.0, 20.0, 20.0))\n'
+          '└SemanticsNode#10(stale, owner: null, Rect.fromLTRB(10.0, 10.0, 10.0, 10.0))\n',
+    );
+
+    final SemanticsNode child3 = new SemanticsNode()
+      ..rect = new Rect.fromLTRB(0.0, 0.0, 0.0, 0.0);
+    child3.addChildren(<SemanticsNode>[
+      new SemanticsNode()..rect = new Rect.fromLTRB(20.0, 0.0, 20.0, 0.0),
+      new SemanticsNode(),
+    ]);
+    child3.finalizeChildren();
+
+    final SemanticsNode rootComplex = new SemanticsNode();
+    rootComplex.addChildren(<SemanticsNode>[child1, child2, child3]);
+    rootComplex.finalizeChildren();
+
+    expect(
+      rootComplex.toStringDeep(childOrder: DebugSemanticsDumpOrder.traversal),
+      'SemanticsNode#15(stale, owner: null, Rect.fromLTRB(0.0, 0.0, 0.0, 0.0))\n'
+      '├SemanticsNode#12(stale, owner: null, Rect.fromLTRB(0.0, 0.0, 0.0, 0.0))\n'
+      '│├SemanticsNode#14(Rect.fromLTRB(0.0, 0.0, 0.0, 0.0))\n'
+      '│└SemanticsNode#13(stale, owner: null, Rect.fromLTRB(20.0, 0.0, 20.0, 0.0))\n'
+      '├SemanticsNode#10(stale, owner: null, Rect.fromLTRB(10.0, 10.0, 10.0, 10.0))\n'
+      '└SemanticsNode#9(stale, owner: null, Rect.fromLTRB(20.0, 20.0, 20.0, 20.0))\n',
+    );
+
+    expect(
+      rootComplex.toStringDeep(childOrder: DebugSemanticsDumpOrder.inverseHitTest),
+      'SemanticsNode#15(stale, owner: null, Rect.fromLTRB(0.0, 0.0, 0.0, 0.0))\n'
+      '├SemanticsNode#9(stale, owner: null, Rect.fromLTRB(20.0, 20.0, 20.0, 20.0))\n'
+      '├SemanticsNode#10(stale, owner: null, Rect.fromLTRB(10.0, 10.0, 10.0, 10.0))\n'
+      '└SemanticsNode#12(stale, owner: null, Rect.fromLTRB(0.0, 0.0, 0.0, 0.0))\n'
+      ' ├SemanticsNode#13(stale, owner: null, Rect.fromLTRB(20.0, 0.0, 20.0, 0.0))\n'
+      ' └SemanticsNode#14(Rect.fromLTRB(0.0, 0.0, 0.0, 0.0))\n',
+    );
+  });
+
+  test('debug properties', () {
+    expect(
+      new SemanticsNode().toStringDeep(),
+      'SemanticsNode#16(Rect.fromLTRB(0.0, 0.0, 0.0, 0.0))\n',
+    );
+
+    final SemanticsNode allProperties = new SemanticsNode()
+      ..rect = new Rect.fromLTWH(50.0, 10.0, 20.0, 30.0)
+      ..mergeAllDescendantsIntoThisNode = true
+      ..transform = new Matrix4.translation(new Vector3(10.0, 10.0, 0.0))
+      ..wasAffectedByClip = true
+      ..addAction(SemanticsAction.scrollUp)
+      ..addAction(SemanticsAction.longPress)
+      ..addAction(SemanticsAction.showOnScreen)
+      ..isChecked = false
+      ..isSelected = true
+      ..label = "Use all the properties";
+    expect(
+      allProperties.toStringDeep(),
+      'SemanticsNode#17(stale, owner: null, leaf merge, Rect.fromLTRB(60.0, 20.0, 80.0, 50.0), clipped, [longPress, scrollUp, showOnScreen], selected, "Use all the properties")\n',
```

Comment:
It seems a little confusing that `isSelected` is reported as `selected` on a `SemanticsNode` as seen here and as `[isSelected]` on a `SemanticsData` item.

Reply:
This was inconsistent in the previous output but there weren't many tests to show what it looked like.
Previously the output was
'SemanticsFlags.isSelected' vs 'selected'
for the two different toString methods.
Let me know what you'd prefer.
It could make sense to show the property name for the flags and actions. Then the output would change to:
flags: [isSelected] vs 
selected

Reply:
I think I would prefer if the output is always the name of the flag.

Reply:
@Hixie do you have a preference between 'selected' and 'isSelected'
Using 'isSelected' here would not be consistent with how the flag like property names are displayed for the Widget and RenderObject trees but would be consistent with SemanticsFlags. I'm fine either way.

Reply:
After looking over this again, I actually changed my mind and think that "selected" is the better choice. This becomes clear when looking at the other flag "IsChecked" - we want that one to print something if it is not present if the node is in theory checkable. To be consistent with that, we should leave it as you have it right now.